cmd/state-offline-uninstaller/uninstall.go

```diff
@@ -54,11 +67,48 @@ func newParams() *Params {
 	return &Params{path: "/tmp"}
 }
 
+func (r *runner) Event(eventType string, installerDimensions *dimensions.Values) {
+	r.analytics.Event(ac.CatRuntimeUsage, eventType, installerDimensions)
+}
+
+func (r *runner) EventWithLabel(eventType string, msg string, installerDimensions *dimensions.Values) {
+	r.analytics.EventWithLabel(ac.CatRuntimeUsage, eventType, msg, installerDimensions)
+}
+
+func (r *runner) handleFailure(err error, msg string, installerDimensions *dimensions.Values) error {
+	r.EventWithLabel("failure", msg, installerDimensions)
+	return errs.Wrap(err, msg)
+}
+
+func (r *runner) handleFailureNewErr(msg string, installerDimensions *dimensions.Values) error {
+	r.EventWithLabel("failure", msg, installerDimensions)
+	return errs.New(msg)
+}
+
 func (r *runner) Run(params *Params) error {
 	licenseFilePath := filepath.Join(params.path, licenseFileName)
+	installerConfigPath := filepath.Join(params.path, installerConfigFileName)
+
+	configData, err := os.ReadFile(installerConfigPath)
+	if err != nil {
+		return errs.Wrap(err, "Failed to read config file, is this an install directory?")
+	}
+	installerConfig := InstallerConfig{}
+
+	if err := json.Unmarshal([]byte(configData), &installerConfig); err != nil {
+		return errs.Wrap(err, "Failed to decode config file")
+	}
+
+	installerDimensions := &dimensions.Values{
+		ProjectID: installerConfig.ProjectID,
+		CommitID:  installerConfig.CommitID,
+		Trigger:   p.StrP(target.TriggerCliOfflineUninstaller.String()),
+	}
+	r.analytics.Event(ac.CatRuntimeUsage, "start", installerDimensions)
```

Comment:
Multiple issues here:

* CatRuntimeUsage does not have a "start" action, see https://docs.google.com/document/d/12-IHGV6iBGE12p61cwKvc16YzahplKxJYSh30HKaBRk/edit#heading=h.u1lsd920bwhf
* Why are we tracking an uninstall as runtime usage? I imagine you would want to fire whatever event you guys landed on for deducting the runtime usage.
* This is currently running before the validation of the install directory (line 114).

Reply:
Note that for just tracking the runtime usage you do not need to make any special calls to the analytics mechanic, you just need to ensure that you pass it a valid instance of the synchronous analytics client here: https://github.com/ActiveState/cli/blob/ec562261f7b30dcebad79f30ab327f217090ebdc/cmd/state-offline-installer/install.go#L150

Reply:
Offline installer now has its own category

Reply:
That only addresses the first point.

Reply:
2nd point - we now send both a category=runtime-use, action=delete and a category=offline-installer, action=success, trigger=offline-uninstaller event on successful uninstall as discussed
3rd point - this event is to denote the user initiating the uninstall process, to be later correlated with the success or failure event that indicates the final outcome

Reply:
3rd point is still a problem imo. Because the curated error that is handled on line 114 will never be able to fire because it will run into the new uncurated error you're introducing in this PR.

I don't feel very strongly about this, I'm just pointing out that the code as written effectively makes the error handling on line 114 pointless.